pkg/api/types.go

```diff
@@ -320,8 +320,81 @@ type Result struct {
 
 	// ImageID describes resulting image ID.
 	ImageID string
+
+	// BuildInfo holds information about the result of a build.
+	BuildInfo []*BuildStepInfo
```

Comment:
it's not just failure reasons, but a more generic build information. it's one of the reasons why i didn't name it buildfailurereasons or something similar. we could attach different data to it, such as the timing of functions (sorry to bring it up again :)) or whatever else we need.